src/Coriolis/spherical_coriolis.jl

```diff
@@ -1,25 +1,27 @@
 using Oceananigans.Grids: LatitudeLongitudeGrid, OrthogonalSphericalShellGrid, peripheral_node, φnode
-using Oceananigans.Operators: Δx_qᶜᶠᶜ, Δy_qᶠᶜᶜ, Δxᶠᶜᶜ, Δyᶜᶠᶜ, hack_sind
+using Oceananigans.Operators: Δx_qᶜᶠᶜ, Δy_qᶠᶜᶜ, Δxᶠᶜᶜ, Δyᶜᶠᶜ, hack_sind, hack_cosd
 using Oceananigans.Advection: EnergyConserving, EnstrophyConserving
 using Oceananigans.BoundaryConditions
 using Oceananigans.Fields
 using Oceananigans.AbstractOperations: KernelFunctionOperation
 using Oceananigans.ImmersedBoundaries
 
 """
-    struct HydrostaticSphericalCoriolis{S, FT} <: AbstractRotation
+    struct SphericalCoriolis{S, FT} <: AbstractRotation
 
 A parameter object for constant rotation around a vertical axis on the sphere.
 """
-struct HydrostaticSphericalCoriolis{S, FT} <: AbstractRotation
+struct SphericalCoriolis{H, S, FT} <: AbstractRotation
     rotation_rate :: FT
     scheme :: S
+    HydrostaticCoriolis :: H
```

Comment:
Will this work? Whether or not we are using the hydrostatic approximation has to be encoded in a type, which is known at compile time.

Reply:
Thanks @glwagner .  I'll give this a try  and get back to you when I have something together.